nasa-gibs/worldview
# WV-2463, WV-2460, WV-2400, WV-2462: Granule layer fixes

## Description

WV-2463: Filter out granules that do not have associated GIBS imagery based on longitudinal bounds

WV-2460: Fix date parsing when displaying available imagery date ranges

WV-2400: Use a layer's visualization metadata from GIBS to determine what collection to query CMR for

WV-2462: Footprints don't always show on page load


## How To Test

You will need to have the GIBS layer config repository cloned locally to test.  You will also need to have docker running an httpd container with the layer metadata directory mounted.

```
// get the httpd image
docker pull httpd

// start a container named gibs-config which serves the contents of the layers-config repo on localhost:8080
docker run --name gibs-config -d -p 8080:80 -v  /<your_path>/layers-config:/usr/local/apache2/htdocs httpd
```

* Checkout `granule-configs`
* Change the `features.json` `vismetadata` `url` property to `"http://localhost:8080/layer-metadata/v1.0/"`
* `npm run build`
* Checkout this branch
* `npm run watch`

WV-2463:
* Load a granule layer
* Step around through time and confirm that granule footprints only show over imagery within the defined longitudinal bounds (-65 to 65 in geographic)

WV-2460:
* Confim that the start and end date availability as shown in the layer info panel matches what is seen when setting the timeline to those datetimes.  
** (e.g. if the end time is 2022-10-12T00:48Z then set the timeline to this and confirm that imagery still shows but not after this time)

WV-2400:
* If granules are loading, this is working properly

WV-2462:
* Open a link that should have granule imagery visible ([like this](http://localhost:3000/?z=4&ics=true&ici=5&icd=10&l=VIIRS_NOAA20_CorrectedReflectance_TrueColor_Granule,Reference_Labels_15m(hidden),Reference_Features_15m(hidden),Coastlines_15m,VIIRS_NOAA20_CorrectedReflectance_TrueColor(hidden),VIIRS_SNPP_CorrectedReflectance_TrueColor(hidden),MODIS_Terra_CorrectedReflectance_TrueColor(hidden)&lg=true&t=2022-10-10-T12%3A45%3A00Z))
* When page loads, confirm granule footprints are visible on hover
* If possible, try in a new incognito window or other various browsers
